sas-viya/azuredeploy.json

```diff
@@ -101,20 +117,18 @@
     "VirtualNetworkApplicationGatewaySubnet": "[concat(variables('VirtualNetworkName'), '_applicationGateway')]",
     "VirtualNetworkApplicationGatewaySubnetCIDR": "10.0.129.0/24",
 
-    "LoadBalancerName": "PrimaryViyaLoadbalancer",
-    "LoadBalancerNetworkSecurityGroup": "[concat(variables('LoadBalancerName'), '_NetworkSecurityGroup')]",
-    "AnsibleController_name": "Ansible",
-    "CASController_name": "controller",
-    "Services_name": "services",
+
     "DiagnosticStorageGroupName": "[toLower(concat('rg4diag', variables('resourceGroupUniqueString')))]",
+    "DiagnosticStorageGroupName_long": "[concat('Microsoft.Storage/storageAccounts/', variables('DiagnosticStorageGroupName'))]",
     "AzureFilesViyaShare": "viyashare",
     "PrimaryUserName": "vmuser",
     "ExtensionAnsibleURI": "[concat(parameters('_artifactsLocation'), 'nestedtemplates/run_ansible_script.json', parameters('_artifactsLocationSasToken'))]",
     "NetworkCreateTemplateURI": "[concat(parameters('_artifactsLocation'), 'nestedtemplates/createNetworkSubtemplate.json', parameters('_artifactsLocationSasToken'))]",
     "AlternateDomain": "[concat('viya-', subscription().subscriptionId, '-', variables('resourceGroupUniqueString'))]",
     "DomainActual": "[ substring(variables('AlternateDomain'),0,if(lessOrEquals(length(variables('AlternateDomain')),60), length(variables('AlternateDomain')), 60))]",
     "Base64AdminPass": "[base64(parameters('SASAdminPass'))]",
-    "Base64UserPass": "[base64(parameters('SASUserPass'))]"
+    "Base64UserPass": "[base64(parameters('SASUserPass'))]",
+    "DUMMY_LICENSE_STRING": "<REPLACE THIS WITH A LINK TO YOUR LICENSE>"
```

Comment:
can we make this a parameter?  And just name it licenseString (or something that doesn't say "dummy" - you can make the defaultValue whatever, e.g.

This is a dummy license, you will need to provide a valid license to run this workload in production

Or words to that effect...

Reply:
This variable is just the value compared against the valued in parameter 'DeploymentDataLocation'. I made it a variable because we do the comparison multiple times.

Reply:
In theory, the only way they run into this string is if they submit the azuredeploy.parameters.json, which also has replace values for your CI system. This value causes it to not do a full deploy, but just a resources only test deploy.

Reply:
Here's what I think would make it a bit cleaner:

- Set the defaultValue of the DeploymentDataLocation to an empty string
- remove the variable
- in your conditional checks for a valid license use:
    if(empty(parameters('DeploymentDataLocation;))...

The concern I have with the way it is written is that it looks like the user has to update the variable in the template to provide a valid license (and then the variable is still called "dummy license").  Using only the parameter makes it a bit clearer as to your intention for the users.

Reply:
So, the issue we have with setting a default of "" is that that will cause it to allow an empty string submission of the DeploymentDataLocation on the Web UI without complaint. On the cli, it will also not fail, but run a full resource deploy and put in the outputs that this was a dummy deploy, which is not a great experience. Since you need a license for the deployment to fully release sas, we don't want to give the impression that you can submit without it, so we tried to make the dummy value suggest you needed to replace it with a valid license in the off chance that they encountered it.

Reply:
So you just want to make the parameter required?  (which it isn't if you have a defaultValue defined) - if so, then just use the empty string in the param file for the quickstart... then the UI will require it (since it doesn't use the param file)

Reply:
Updated as requested.